src/Reflection/RequireExtension/RequireExtendsMethodsClassReflectionExtension.php

```diff
@@ -0,0 +1,73 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Reflection\RequireExtension;
+
+use PHPStan\Analyser\OutOfClassScope;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\MethodsClassReflectionExtension;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\VerbosityLevel;
+
+class RequireExtendsMethodsClassReflectionExtension implements MethodsClassReflectionExtension
+{
+
+	/** @var array<string, array<string, true>> */
+	private array $inProcess = [];
+
+	public function hasMethod(ClassReflection $classReflection, string $methodName): bool
+	{
+		return $this->findMethod($classReflection, $methodName) !== null;
+	}
+
+	public function getMethod(ClassReflection $classReflection, string $methodName): MethodReflection
+	{
+		$method = $this->findMethod($classReflection, $methodName);
+		if ($method === null) {
+			throw new ShouldNotHappenException();
+		}
+
+		return $method;
+	}
+
+	private function findMethod(ClassReflection $classReflection, string $methodName): ?MethodReflection
+	{
+		if (!$classReflection->isInterface()) {
+			return null;
```

Comment:
This is a point for the rules - `@...-require-extends` can only be above an interface.

Reply:
so do you mean which should still try to infer the property across the type hierarchy, even though the `require-extends` is placed on a invalid location?

Reply:
Sorry, I don't get it. This comment was just a reminder for you to check whether `require-extends` is always above an interface.

Reply:
rules will be topic for a followup PR. thanks for noticing.